### PR TITLE
Fixed sending fortran ordered arrays

### PR DESCRIPTION
**Pull request by david-zwicker**
This PR fixes a bug where data of Fortran ordered arrays was transposed.

### CODE REVIEW

**Comment by david-zwicker** [comment]
Apparently, there are still issues with the code style. I don't fully understand the problem, though.

**Comment by slayoo** [comment]
closing and reopening to run with updated pylint workflow file that requires older version of pylint

**Comment by slayoo** [comment]
seems the problem is a regression in pylint introduced in the recently released 2.15.0

File: numba_mpi/mpi.py [hunk 1]
```diff
@@ -196,7 +196,7 @@ def recv(data, source, tag):
 
     buffer = (
         data if data.flags.c_contiguous
-        else np.empty_like(data)
+        else np.empty(data.shape, data.dtype)
```
Comment (slayoo) [inline numba_mpi/mpi.py:199]: would `np.empty_like(data, order='C')` be more readable?

Comment (david-zwicker) [reply numba_mpi/mpi.py:199]: Yes, I think this would be more expressive.

**Comment by slayoo** [comment]
no luck, Numba does not support the `order` kw arg to `empty_like` :(
(https://numba.pydata.org/numba-doc/dev/reference/numpysupported.html)
reverting last change...

**Pull request merged**